graphstorm-processing/graphstorm_processing/graph_loaders/dist_heterogeneous_loader.py

```diff
@@ -2352,6 +2387,14 @@ def multinomial_sample(label_col: str) -> Sequence[int]:
         int_group_df = input_df.select(
             split_group(input_col).alias(DATA_SPLIT_SET_MASK_COL), *input_df.columns
         )
+        # For homogeneous graph, reverse edge will not have masks
+        if HOMOGENEOUS_REVERSE_COLUMN_FLAG in int_group_df.columns:
+            int_group_df = int_group_df.withColumn(
+                DATA_SPLIT_SET_MASK_COL,
+                when(
+                    F.col(HOMOGENEOUS_REVERSE_COLUMN_FLAG), F.col(DATA_SPLIT_SET_MASK_COL)
+                ).otherwise(F.lit([0, 0, 0])),
+            )
```

Comment:
Can you explain what happens downstream? Why exclude all reverse edges from train/test/val?

Reply:
The idea here is if it is a reversed edge, we do not assign masks to the reverse edge. We only expect we are assigning masks to the original edges. Reversed edges are only for message passing, we do not want to extend the masked edge.